api/v1alpha1/trafficmanagerbackend_types.go

```diff
@@ -0,0 +1,162 @@
+	Properties *TrafficManagerEndpointProperties `json:"properties,omitempty"`
+}
+
+// ProfileRef is a reference to a trafficManagerProfile object.
+type ProfileRef struct {
+	// Name is the name of the referenced trafficManagerProfile.
+	Name string `json:"name"`
+	// Namespace is the namespace where the referenced trafficManagerProfile is located.
+	// If the namespace is omitted, it is assumed to be the same namespace as the trafficManagerBackend.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+}
+
+// TrafficManagerBackendRef is the reference to the endpoint(s).
+// In the multi-cluster scenario, it supports ServiceImport as the endpoint reference, representing a group of endpoints.
+type TrafficManagerBackendRef struct {
+	// Name is the reference to the ServiceImport in the same namespace of Traffic Manager Profile.
+	// +required
+	Name string `json:"name"`
+}
+
+// TrafficManagerEndpointProperties defines the properties of the Traffic Manager endpoint.
+// For now, only AzureEndpoints are supported.
+// It assumes the service is exposed via PublicIPAddress in Azure and must have a DNS name assigned.
+// Note: To use Traffic Manager with endpoints from other subscriptions, the controller needs to have read access to
+// the endpoint.
+type TrafficManagerEndpointProperties struct {
+	// The total weight of endpoints behind the serviceImport when using the 'Weighted' traffic routing method.
+	// Possible values are from 1 to 1000.
+	// It is required when the routing method is 'Weighted'.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=1000
+	// For example, if there are two clusters exporting the service via public ip, each public ip will be configured
+	// as "Weight"/2.
+	Weight *int64 `json:"weight,omitempty"`
+
+	// The fully-qualified DNS name or IP address of the endpoint.
+	// READ-ONLY; populated in the status.
+	// +optional
+	Target *string `json:"target,omitempty"`
+}
+
+// AcceptedEndpoint is the Azure Traffic Manager endpoint which is successfully accepted under the traffic manager Profile.
+type AcceptedEndpoint struct {
+	// Name is a unique Azure Traffic Manager endpoint name generated by the controller.
+	// +required
+	Name *string `json:"name"`
+
+	// +optional
+	Properties *TrafficManagerEndpointProperties `json:"properties,omitempty"`
+
+	// Cluster is where the endpoint is exported from.
+	// +optional
+	Cluster *ClusterStatus `json:"cluster,omitempty"`
+}
+
+type TrafficManagerBackendStatus struct {
+	// Endpoints contains a list of accepted Azure endpoints which are created or updated under the traffic manager Profile.
+	// +optional
+	Endpoints []AcceptedEndpoint `json:"endpoints,omitempty"`
+
+	// Current service state
+	// +optional
+	// +patchMergeKey=type
+	// +patchStrategy=merge
+	// +listType=map
+	// +listMapKey=type
+	Conditions []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Comment:
Are the conditions per endpoint or per backend? How can users observe which endpoints are not accepted yet?

Reply:
The condition is per backend. The "Endpoints" array is to show the accepted endpoints.